src/platform/sound/manager.js

```diff
@@ -125,87 +92,41 @@ class SoundManager extends EventHandler {
-                    this._context.onstatechange = function () {
-
-                        // explicitly call .resume() when previous state was suspended or interrupted
-                        if (self._unlocked && !self._selfSuspended && self._context.state !== CONTEXT_STATE_RUNNING) {
-                            self._context.resume().then(() => {
-                                // no-op
-                            }, (e) => {
-                                Debug.error(`Attempted to resume the AudioContext on onstatechange, but it was rejected`, e);
-                            }).catch((e) => {
-                                Debug.error(`Attempted to resume the AudioContext on onstatechange, but threw an exception`, e);
-                            });
-                        }
-                    };
-                }
+        if (!this._context && this.AudioContext) {
+            this._context = new this.AudioContext();
+
+            if (this._context.state !== CONTEXT_STATE_RUNNING) {
+                this._registerUnlockListeners();
             }
         }
 
         return this._context;
     }
 
     suspend() {
-        this._selfSuspended = true;
-
-        if (this.suspended) {
-            // already suspended
-            return;
+        if (!this._userSuspended) {
+            this._userSuspended = true;
+            if (this._context && this._context.state === CONTEXT_STATE_RUNNING) {
+                this._suspend();
+            }
         }
-
-        this.fire('suspend');
     }
 
     resume() {
-        this._selfSuspended = false;
-
-        // cannot resume context if it wasn't created yet or if it's still locked
-        if (!this._context || (!this._unlocked && !this._unlocking)) {
-            return;
-        }
-
-        // @ts-ignore 'interrupted' is a valid state on iOS
-        if (this._context.state === CONTEXT_STATE_INTERRUPTED) {
-            // explictly resume() context, and only fire 'resume' event after context has resumed
-            this._context.resume().then(() => {
-                this.fire('resume');
-            }, (e) => {
-                Debug.error(`Attempted to resume the AudioContext on SoundManager.resume(), but it was rejected`, e);
-            }).catch((e) => {
-                Debug.error(`Attempted to resume the AudioContext on SoundManager.resume(), but threw an exception`, e);
-            });
-        } else {
-            this.fire('resume');
+        if (this._userSuspended) {
+            this._userSuspended = false;
+            if (this._context && this._context.state !== CONTEXT_STATE_RUNNING) {
+                this._resume();
+            }
         }
     }
 
     destroy() {
-        this._removeUserInputListeners();
-
         this.fire('destroy');
 
-        if (this._context && this._context.close) {
-            this._context.close();
+        if (this._context) {
+            this._removeUnlockListeners();
+            this?._context?.close();
```

Comment:
But isn't `this._context` guaranteed to be valid by this point?

Reply:
Not if AudioContext doesn't exist, like in engine tests :)

Reply:
I'm confused. Then the `if` condition would have failed, no?

Reply:
Oic you're referring to the first ? on this line. Yeah I can remove it. The old code tested for presence of context.close before invoking it, so the second ? should probably stay.

Reply:
Done, thanks!